Media Picker 3 documentation (#3106)
umbraco/UmbracoDocs: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Media-Picker-3/index.md

### Accepted types

Use setting to limit the picker to only select Media Items of these types.

### Pick multiple items

Use this setting to enable the property to contain multiple items, when this is turned on the property editor returns a `IEnumerable<MediaWithCrops>`. You can still set the Maximum Amount to 1, do so if you like to retrive a collection but only allowing the Content Editors to select one Media Item.

### Amount

Use this setting to enforce a certain minimum or maximum amount of selected Media Items, the maximum setting is unuseable when multiple items is disabled.

### Start node

Limit the picker to only work in a certain part of the Media Tree.

### Ignorer user start nodes

Use setting to overrule user permissions, to enable any user of this property to pick any Media Item of the choosen Start node.

When this setting is enabled, a user who doesn't normally have access to the media selected as "Start Node" (/Design in this case), can access the media when using this particular Media Picker. If no Start node has been defined for this property any content can be viewed and selected of this property.

### Image Croppings

> Define local image croppings, this data will stay locally, meaning that this will be stored on the document in this property.

Reviewer: I'm having a hard time understanding what is meant by this sentence.

Would you be able to help me understand?

Author: The existing Image Cropper feature is stored together with the Image-file in the Image Cropper Property Editor. Default is that its used on the Media Type 'Image'. When setting Image Crops on the Image Cropper, they are then shared by all usages of that Media Item. aka. Crops made on Photo of Umbraco HQ, will be the same everywhere that Media Item is used.

The Local Image Crops Feature of MediaPicker 3, is a feature where the crops are stored together with the pick, not on the Media Item itself. Enabling differentiation, as each media item will have custom crops for each usage.

I did reformulate a bit, but please help me out, its rather complex to get everyone's understanding right :-)

Reviewer: Ah! Alright. I understand it now. I like the new wording you've added 💪 Let's stick with that.

## Content Example

![Media Picker Content](images/Media-Picker3-Content.jpg)

## MVC View Example

### Multiple enabled without Modelsbuilder

```csharp
@{
    var typedMultiMediaPicker = Model.Value<IEnumerable<MediaWithCrops>>("medias");
    foreach (var entry in typedMultiMediaPicker)
    {
        <img src="@entry.MediaItem.Url" style="width:200px"/>
    }
}
```

### Multiple enabled with Modelsbuilder

```csharp
@{
    var typedMultiMediaPicker = Model.Medias;
    foreach (var entry in typedMultiMediaPicker)
    {
        <img src="@entry.MediaItem.Url" style="width:200px" />
    }
}
```

> ## Multiple disabled without Modelsbuilder

Reviewer: ```suggestion
### Multiple disabled without Modelsbuilder
```

```csharp
@{
    var typedMediaPickerSingle = Model.Value<MediaWithCrops>("media");
    if (typedMediaPickerSingle != null)
    {
        <img src="@typedMediaPickerSingle.MediaItem.Url" style="width:200px" alt="@typedMediaPickerSingle.MediaItem.Value("alt")" />
    }
}
```

> ## Multiple disabled with Modelsbuilder

Reviewer: ```suggestion
### Multiple disabled with Modelsbuilder
```

```csharp
@{
    var typedMediaPickerSingle = Model.Media;
    if (typedMediaPickerSingle is MediaWithCrops mediaEntry)
    {
        <img src="@mediaEntry.MediaItem.Url" style="width:200px"/>
    }
}
```

## Using local crops

> Local image croppings are stored and retrived different than global crops, below is two examples of how to retrive local crops.

Reviewer: ```suggestion
Local image croppings are stored and retrieved differently than global crops. Below are two examples of how to retrieve local crops.
```

### Using GetLocalCropUrl

```csharp
@{
    foreach (var entry in Model.Medias)
    {
        <img src="@entry.GetLocalCropUrl("cropAlias")"/>
    }
}
```

### Using UrlHelper

```csharp
@{
    foreach (var entry in Model.Medias)
    {
        <img src="@Url.GetCropUrl(entry.LocalCrops, "cropAlias")"/>
    }
}
```

## Use global crops

Global image croppings are crops stored on the Media Item, by the Property Editor `Image Cropper`, making it shared between all usages of the media Item.

The global crops are configured on the DataType of the `umbracoFile` property on the Media Type `Image`

[Read about the Image Cropper here](../Image-Cropper/index.md)

### Using GetCropUrl

```csharp
@{
    foreach (var entry in Model.Medias)
    {
        <img src="@entry.MediaItem.GetCropUrl("cropAlias")"/>
    }
}
```

### Using UrlHelper

```csharp
@{
    foreach (var entry in Model.Medias)
    {
        <img src="@Url.GetCropUrl(entry.MediaItem, "cropAlias")"/>
    }
}

## Add values programmatically

> ```This is not part of the current Media Picker v3, feel free to provide such.```

Reviewer: What exactly does this part mean? 🤔
Does it mean that it is not possible to add values to this property programmatically?

Author: I just don't know how that's done, or whether more code needs to be made to enable such. But its part of MP2, so i thought to point that out.

Reviewer: Good point! I've removed it for now. I will add it to the Docs issue tracker, in case someone in the community wants to help out with adding the section.

### Image Croppings

Define local image croppings, this data will stay locally, meaning that this will be stored on the document in this property.

## Content Example

![Media Picker Content](images/Media-Picker3-Content.jpg)

## MVC View Example

### Multiple enabled without Modelsbuilder

```csharp
@{
    var typedMultiMediaPicker = Model.Value<IEnumerable<MediaWithCrops>>("medias");
    foreach (var entry in typedMultiMediaPicker)
    {
        <img src="@entry.MediaItem.Url" style="width:200px"/>
    }
}
```

### Multiple enabled with Modelsbuilder

```csharp
@{
    var typedMultiMediaPicker = Model.Medias;
    foreach (var entry in typedMultiMediaPicker)
    {
        <img src="@entry.MediaItem.Url" style="width:200px" />
    }
}
```

## Multiple disabled without Modelsbuilder

```csharp
@{
    var typedMediaPickerSingle = Model.Value<MediaWithCrops>("media");
    if (typedMediaPickerSingle != null)
    {

>         <p>@typedMediaPickerSingle.MediaItem.Url</p>

Reviewer 2: I think we should remove the paragraph with url
```suggestion

```

        <img src="@typedMediaPickerSingle.MediaItem.Url" style="width:200px" alt="@typedMediaPickerSingle.MediaItem.Value("alt")" />
    }
}
```

## Multiple disabled with Modelsbuilder

```csharp
@{

>     var typedMediaPickerSingle = Model.FeaturedBanner;

Reviewer 2: I think we should use the same alias as the example above
```suggestion
    var typedMediaPickerSingle = Model.Media;
```

    if (typedMediaPickerSingle is MediaWithCrops mediaEntry)
    {

>         <p>@mediaEntry.MediaItem.Url</p>

Reviewer 2: I think we should remove the paragraph
```suggestion

```

        <img src="@mediaEntry.MediaItem.Url" style="width:200px"/>
    }
}
```

## Using local crops

Local image croppings are stored and retrived different than global crops, below is two examples of how to retrive local crops.

### Using GetLocalCropUrl

```csharp
@{
    foreach (var entry in Model.Medias)
    {
        <img src="@entry.GetLocalCropUrl("cropAlias")"/>
    }
}
```

### Using UrlHelper

```csharp
@{
    foreach (var entry in Model.Medias)
    {
        <img src="@Url.GetCropUrl(entry.LocalCrops, "cropAlias")"/>
    }
}
```

## Use global crops

> Global image croppings are croppes stored on the Media Item, by the Property Editor `Image Cropper`, making it shared between all usages of the media Item.

Reviewer 2: ```suggestion
Global image croppings are crops stored on the Media Item, by the Property Editor `Image Cropper`, making it shared between all usages of the media Item.
```
